intermittent panics with TestStartBuildRunFollowLog

/kind bug

still seeing some intermittent panics with the multi-threaded `TestStartBuildRunFollowLog` unit test I recently added

```
=== RUN   TestStartBuildRunFollowLog
    run_test.go:173: Run initialization did not complete in time
--- FAIL: TestStartBuildRunFollowLog (3.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1b9e6cd]

goroutine 9 [running]:
testing.tRunner.func1.2({0x1c9cdc0, 0x2c0bd70})
	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1209 +0x36c
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1212 +0x3b6
panic({0x1c9cdc0, 0x2c0bd70})
	/opt/hostedtoolcache/go/1.17.1/x64/src/runtime/panic.go:1047 +0x266
github.com/shipwright-io/cli/pkg/shp/cmd/build.(*RunCommand).onEvent(0xc0001901c0, 0xc000346400)
	/home/runner/work/cli/cli/pkg/shp/cmd/build/run.go:128 +0xe8d
github.com/shipwright-io/cli/pkg/shp/cmd/build.TestStartBuildRunFollowLog(0xc0003e6b60)
	/home/runner/work/cli/cli/pkg/shp/cmd/build/run_test.go:178 +0x1508
testing.tRunner(0xc0003e6b60, 0x1ed3f18)
	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1259 +0x230
created by testing.(*T).Run
	/opt/hostedtoolcache/go/1.17.1/x64/src/testing/testing.go:1306 +0x727
FAIL	github.com/shipwright-io/cli/pkg/shp/cmd/build	3.211s
```

--- a/pkg/shp/cmd/build/run_test.go
+++ b/pkg/shp/cmd/build/run_test.go
@@ -162,7 +162,9 @@ func TestStartBuildRunFollowLog(t *testing.T) {
 		// the init to occur.
 		cmd.watchLock.Lock()
 		err := wait.PollImmediate(1*time.Second, 3*time.Second, func() (done bool, err error) {
-			if cmd.pw != nil {
+			// check any of the vars on RunCommand that are used in onEvent and make sure they are set;
+			// we are verifying the initialization done in Run() on RunCommand is complete
+			if cmd.pw != nil && cmd.ioStreams != nil && cmd.shpClientset != nil {
 				cmd.watchLock.Unlock()
 				return true, nil
 			}

